### DISCUSSION TITLE
File Manager: Shorting folders as if they were files

### DISCUSSION THREAD

**Discussion by es20490446e** [linuxdeepin/developer-center#3039] [Features Request & Ideas | 特性请求 & 头脑风暴]
The file manager shorts folders before files. I think it will be cool having the option to short them as if they were regular files, it allows more interesting organisations.

For example if I have a folder with my projects:
P1
P2
P3
P4

And P3 only consist of a plain text file, while the rest of multiple files, I can choose to put P3.txt right away on the project folder, and the shorting remain consistent:
P1
P2
P3.txt
P4

It's less common, but I believe that conceptually it's easier to understand. That's why the Linux terminal shorts like this, while graphical tools do the opposite most likely for consistency with Windows.

**Comment by keybreak** [comment]
I personally would absolutely go against this sorting, as this is one of the things that makes Deepin file manager very usable.

@BLumia 
Please make this feature optional in Settings, if it will be implemented, but keep defaults as they are now.

**Comment by meiyixiang** [comment]
Hello, we probably won't do that, we'll have our own rules for sorting, for example, type, time, size, but we won't give you any kind of directional sorting
